.github/workflows/lxd.yaml

```
@@ -0,0 +1,49 @@
+name: LXD
+
+on:
+  push:
+    branches: [main]
+    paths-ignore: ['**.md', '**.erb', 'schemas/*']
+  pull_request:
+    type: [opened, reopened, edited]
+    paths-ignore: ['**.md', '**.erb', 'schemas/*']
+
+jobs:
+  lxd:
+    name: LXD
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Add user to LXD group
+        run: |
+          sudo usermod -aG lxd $USER
+          group="$(id -g)"
+          newgrp lxd
+          newgrp $group
+      - name: Setup Ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: '2.7.x'
+      - name: Install bundler
+        run: |
+          gem install bundler
+          bundle config path vendor/bundle
+      - name: Cache gems
+        id: cache
+        uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('lib/bolt/version.rb') }}-${{ hashFiles('Gemfile') }}-${{ hashFiles('bolt.gemspec') }}
+      - name: Install gems
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: bundle install --jobs 4 --retry 3
+      - name: Update gems
+        if: steps.cache.outputs.cache-hit == 'true'
+        run: bundle update
```

```suggestion
      - name: Setup Ruby
        uses: actions/setup-ruby@v1
        with:
          ruby-version: 2.7
          bundler-cache: true
      - name: Update gems
        run: bundle update
```

`actions/ruby-setup` is deprecated. The new `ruby/setup-ruby` action also takes care of installing bundler and then installing and caching gems.